eng/templates/stages/deploy.yaml

```diff
@@ -203,6 +203,24 @@ stages:
         .\darc\darc.exe get-default-channels --source-repo arcade-services --ci -t "$(GetAuthInfo.FederatedToken)" --bar-uri "$(GetAuthInfo.BarUri)" --debug
       displayName: Test Federated token authentication
 
+    - ${{ if in(variables['Build.SourceBranch'], 'refs/heads/main', 'refs/heads/production') }}:
```

Comment:
Maybe it makes sense to run it regardless of branch? Say, we make a change to something and run the internal pipeline on our dev branch, we would want this executed to test that we did not break the command

Reply:
it won't work if it's not on main or prod, because the build won't get added to darc, so we'll have no bar id

Reply:
And we can't just use some static bard id for a build that already exists and will work?

Reply:
no, because the build will already be in that channel, and you can't publish it twice

Reply:
Hmm, I see. It just feels a bit late if the change causing the break is already in `main` / `prod`

Reply:
I mean it's meant to work this way, like a scenario test. We will never deploy this to prod if main is red